File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/Rerank2Api.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class Rerank2Api {
    private final HttpClient client;
    
    public Rerank2Api(HttpClient client) {
        this.client = client;
    }

    /** Rerank documents */
    public RerankResponse rerank(RerankRequest body) throws Exception {
        return (RerankResponse) client.post(ApiPaths.aiPath("/v1/rerank"), body);
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/Rerank2Api.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class Rerank2Api(private val client: HttpClient) {

    /** Rerank documents */
    suspend fun rerank(body: RerankRequest): RerankResponse? {
        return client.post(ApiPaths.aiPath("/v1/rerank"), body) as? RerankResponse
    }
}
